subsys/dsp/CMakeLists.txt

```diff
@@ -1,8 +1,10 @@
 # Copyright (c) 2022 Google LLC
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_library_named(zdsp)
+if(CONFIG_CMSIS_DSP)
```

Comment:
This is not logically correct. ZDSP is a standalone subsystem independent from the CMSIS-DSP.

There should have been `menuconfig DSP` in `subsys/dsp/Kconfig` to enable DSP subsystem.

Reply:
The plan was to have this condition expanded in the future with ORs when other DSP backends are added. I agree that having DSP as a kconfig symbol is preferable, but it would require a bigger change (CONFIG_DSP=y everywhere), which I thought would be unwanted. Do you prefer that I leave it that way or create a new kconfig symbol?

Reply:
As with any other subsystems we have, the DSP subsystem should have a "root" Kconfig symbol (`CONFIG_DSP`), we forgot to add one in the initial PR due to an oversight.

If you can add one in this PR, that would be great; if not, @yuval or I will look into doing that in a follow up PR.